scripts/test-tls13-ecdsa-brainpool-in-certificate-verify.py

```diff
@@ -0,0 +1,582 @@
+        conversations["check {0} signature {1}".format(
+                      name, result)] = conversation
+
+    # verify that an ECDSA signature with mismatched message hash fails
+    if private_key.curve_name == "BRAINPOOLP256r1":
+        sig_alg = SignatureScheme.ecdsa_brainpoolP384r1tls13_sha384
+        msg_alg = SignatureScheme.ecdsa_brainpoolP256r1tls13_sha256
+    elif private_key.curve_name == "BRAINPOOLP384r1":
+        sig_alg = SignatureScheme.ecdsa_brainpoolP256r1tls13_sha256
+        msg_alg = SignatureScheme.ecdsa_brainpoolP384r1tls13_sha384
+    else:
+        assert private_key.curve_name == "BRAINPOOLP512r1"
+        sig_alg = SignatureScheme.ecdsa_brainpoolP384r1tls13_sha384
+        msg_alg = SignatureScheme.ecdsa_brainpoolP512r1tls13_sha512
+
+    conversation = Connect(hostname, port)
+    node = conversation
+    ciphers = [CipherSuite.TLS_AES_128_GCM_SHA256,
+               CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
+    ext = {}
+    groups = [GroupName.secp256r1,
+              GroupName.brainpoolP256r1tls13,
+              GroupName.brainpoolP384r1tls13,
+              GroupName.brainpoolP512r1tls13]
+    ext[ExtensionType.key_share] = key_share_ext_gen(groups)
+    ext[ExtensionType.supported_versions] = \
+        SupportedVersionsExtension().create([(3, 4), (3, 3)])
+    ext[ExtensionType.supported_groups] = \
+        SupportedGroupsExtension().create(groups)
+    ext[ExtensionType.signature_algorithms] = \
+        SignatureAlgorithmsExtension().create(sig_algs)
+    ext[ExtensionType.signature_algorithms_cert] = \
+        SignatureAlgorithmsCertExtension().create(ECDSA_SIG_ALL + RSA_SIG_ALL)
+    node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
+    node = node.add_child(ExpectServerHello())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectEncryptedExtensions())
+    node = node.add_child(ExpectCertificateRequest())
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectCertificateVerify())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(CertificateGenerator(X509CertChain([cert])))
+    node = node.add_child(CertificateVerifyGenerator(
+        private_key, sig_alg=sig_alg, msg_alg=msg_alg))
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectAlert(
+        AlertLevel.fatal, AlertDescription.decrypt_error))
+    node.add_child(ExpectClose())
+
+    conversations["check ecdsa signature with mismatched hash fails"] = \
+        conversation
+
+
+    # check that fuzzed signatures are rejected
+    if len(private_key) == 256:
+        # bacause of DER encoding of the signature, the mapping between key size
+        # and signature size is non-linear
+        siglen = 70
+    elif len(private_key) == 384:
+        siglen = 103
+    elif len(private_key) == 512:
+        siglen = 135
+    else:
+        assert len(private_key) == 521
+        siglen = 137
```

Comment:
Some associative array here instead of lengths?

Reply:
for just 4 values I don't think it improves readability much...